### PR TITLE
OGM-791 MongoDB Driver 2.13 | MongoDB 3.0 credentials | Fongo update

### PR DESCRIPTION
**Pull request by emmanuelbernard**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -51,7 +51,7 @@
         <skipUnitTests>false</skipUnitTests>
 
         <!-- The following properties configure the embedded MongoDB instance used for tests -->
-        <embeddedMongoDbVersion>2.6.5</embeddedMongoDbVersion>
+        <embeddedMongoDbVersion>3.0.2</embeddedMongoDbVersion>
```
Comment (gunnarmorling) [inline pom.xml]: Should we have another profile for 2.6 so we can run tests against it on CI? It's a more general question I guess, should we only test against latest major versions of backends or also previous ones.

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/AuthenticationMechanismType.java [hunk 2]
```diff
@@ -39,7 +44,25 @@ public MongoCredential createCredential(String username, String databaseName, St
 		public MongoCredential createCredential(String username, String databaseName, String password) {
 			return MongoCredential.createMongoX509Credential( username );
 		}
-	};
+	}
+	,
+	SCRAM_SHA_1 {
```
Comment (gunnarmorling) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/AuthenticationMechanismType.java]: Could you update the docs of `MongoDBProperties.AUTHENTICATION_MECHANISM` to state the new default value? I don't think we need to list the actual values there, a ref to the enum is good enough.

File: pom.xml (hunk 1 above)
Comment (emmanuelbernard) [inline pom.xml]: I'm inclined not to. I did test it manually.

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/AuthenticationMechanismType.java (hunk 2 above)
Comment (emmanuelbernard) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/AuthenticationMechanismType.java]: ok. Which reminds me I did not look at the documentation either

File: pom.xml (hunk 1 above)
Comment (DavideD) [inline pom.xml]: I think as well, we should test only the latest versions. At least for now.
If at some point we receive many requests from the users about using specific versions we might change approach.

**Comment by emmanuelbernard** [comment]
@gunnarmorling Update with the documentation changes.
I also anticipated the `WriteConcern` deprecation of MongoDB driver 3 with a  soft deprecation.

**Comment by gunnarmorling** [comment]
Merged; Thanks.

**Pull request merged**